src/Compilers/Server/VBCSCompiler/VBCSCompiler.cs

```diff
@@ -4,13 +4,33 @@
 using System;
 using System.Collections.Specialized;
 using System.IO;
+using System.Runtime;
 
 namespace Microsoft.CodeAnalysis.CompilerServer
 {
     internal static class VBCSCompiler 
     {
         public static int Main(string[] args)
         {
+            // Enable multi-core JITing
+            // https://blogs.msdn.microsoft.com/dotnet/2012/10/18/an-easy-solution-for-improving-app-launch-performance/
+#if NET46
+            var profileRoot = Path.Combine(
+                Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
+                nameof(VBCSCompiler),
+                "ProfileOptimization");
+            try
+            {
+                Directory.CreateDirectory(profileRoot);
+                ProfileOptimization.SetProfileRoot(profileRoot);
+                ProfileOptimization.StartProfile("Startup.profile");
```

Comment:
Can this succeed but cause downstream failures around file locking?

Reply:
I feel a bit uneasy about dealing with environment/paths/files and ignoring exceptions, but logically this seems the right thing to do here.

Reply:
Even in NGENd quite a few methods must be JIT-ed, so this should have positive impact on start up. I wonder how much faster does it get when compiling HelloWorld? (NGENd or not)

Reply:
Agree we need to pokemon catch here but think we need to log the failure some how. Important if we are ever to diagnose problems with this approach.

Reply:
Also- what happens when a given machine/user have seen multiple compiler versions? 
Do they all share the oldest profile?  - I am not sure how smart LocalApplicationData is. Is it global or has a different location for different users, TS sessions, versions of the binary, etc ...